src/terminal/adapter/adaptDispatch.cpp

```diff
@@ -3145,6 +3145,12 @@ bool AdaptDispatch::_EraseScrollback()
     auto& cursor = textBuffer.GetCursor();
     const auto row = cursor.GetPosition().y;
 
+    // Scroll all the marks up. This will trim ones that are now "outside" the buffer
+    textBuffer.ScrollMarks(-top);
+    // Also clear all the marks below the new viewport position.
+    textBuffer.ClearMarksInRange(til::point{ 0, height },
+                                 til::point{ bufferSize.width, bufferSize.height });
```

Comment:
You could swap these two to reduce the amount of work. First clear everything outside of the `viewport` and then scroll the remaining bits up by `-top`.